app/models/pipeline_run.rb

```diff
@@ -153,38 +157,255 @@ def retry
+    `
+    _stdout, _stderr, _status = Open3.capture3("rm -f #{downloaded_byteranges_path}")
+  end
+
+  def s3_output_for(db_load_command)
+    case db_load_command
+    when "db_load_host_filtering"
+      "#{host_filter_output_s3_path}/#{STATS_JSON_NAME}"
+    when "db_load_alignment"
+      "#{alignment_output_s3_path}/#{output_json_name}"
+    when "db_load_postprocess"
+      "#{postprocess_output_s3_path}/#{TAXID_BYTERANGE_JSON_NAME}"
+    end
+  end
+
+  def output_ready?(db_load_command)
+    file_generated_since_run(s3_output_for(db_load_command))
+  end
+
+  def update_result_status(field, value)
+    status_hash = result_status_hash
+    status_hash[field] = value
+    update(result_status: status_hash.to_json)
+  end
+
+  def result_status_for(field)
+    status_hash = result_status_hash
+    status_hash[field]
+  end
+
+  def result_status_hash
+    result_status ? JSON.parse(result_status) : {}
+  end
+
+  def status_display(h = result_status_hash)
+    # Status display for the frontend.
+    # This function would be simpler if we used active_stage (job monitor),
+    # but we want to use result_status instead (result monitor)
+    # so that it becomes easy to expose availability of specific results
+    # with more granularity later if we desire.
+    if h.empty? || h.values.compact.empty?
+      # No status has been set yet
+      "WAITING"
+    elsif [h["db_load_alignment"], h["db_load_host_filtering"]].include?(STATUS_FAILED)
+      # host-filtering or non-host alignment failed
+      "FAILED"
+    elsif h["db_load_alignment"] == STATUS_LOADED && h["db_load_postprocess"] == STATUS_FAILED
+      # report is ready but postprocessing failed
+      "COMPLETE*"
+    elsif h["db_load_postprocess"] == STATUS_LOADED
+      # postprocessing succeeded
+      "COMPLETE"
+    elsif h["db_load_alignment"] == STATUS_LOADED
+      # alignment succeeded, postprocessing in progress
+      "POST PROCESSING"
+    elsif h["db_load_host_filtering"] == STATUS_LOADED
+      # host-filtering succeeded, alignment in progress
+      "ALIGNMENT"
+    else
+      # host-filtering in progress
+      "HOST FILTERING"
+    end
+  end
+
+  def status_display_pre_result_monitor(run_stages)
+    status_by_loader = {}
+    run_stages.each do |rs|
+      status_by_loader[rs.load_db_command_func] = rs.job_status
+    end
+    status_display(status_by_loader)
+  end
+
+  def status_display_pre_run_stages
+    if %w[CHECKED SUCCEEDED].include?(job_status)
+      'COMPLETE'
+    elsif %w[FAILED ERROR].include?(job_status)
+      'FAILED'
+    elsif %w[RUNNING LOADED].include?(job_status)
+      'IN PROGRESS'
+    else
+      'WAITING'
+    end
+  end
+
+  def check_and_enqueue(db_load_command_name)
+    # TODO: handle case where resque crashes and needs to be restarted. What happens to runs that were queued to load results?
+    if output_ready?(db_load_command_name) && ![STATUS_LOADED, STATUS_LOADING].include?(result_status_for(db_load_command_name))
```

Comment:
I think this is still problematic but we can deal with it later (namely, i think we should come up with the list of files that we want to load rather than still using the stages as the key to monitor.)   The most important thing though is to get rid of the dependency of the stats.json file as the time to "load" the results".

Reply:
OK. Although the result monitor here still uses the load commands we originally wrote for the pipeline_run_stages, it no longer makes any reference to the run stages directly. I can change it so it reads `output_ready?("taxon_counts")` instead of `output_ready?("db_load_alignment")`